pkg/readiness/object_tracker.go

```diff
@@ -56,15 +60,21 @@ type objectTracker struct {
 	populated     bool                      // all expectations have been provided
 	allSatisfied  bool                      // true once all expectations have been satisfied. Acts as a circuit-breaker.
 	kindsSnapshot []schema.GroupVersionKind // Snapshot of kinds before freeing memory in Satisfied.
+	dataFactory   objDataFactory            // Function that creates objData types.  Allows for setting defaults.
 }
 
-func newObjTracker(gvk schema.GroupVersionKind) *objectTracker {
+func newObjTracker(gvk schema.GroupVersionKind, fn objDataFactory) *objectTracker {
+	if fn == nil {
+		fn = objDataFromFlags
+	}
+
 	return &objectTracker{
-		gvk:       gvk,
-		cancelled: make(objSet),
-		expect:    make(objSet),
-		seen:      make(objSet),
-		satisfied: make(objSet),
+		gvk:         gvk,
+		cancelled:   make(objSet),
+		expect:      make(objSet),
+		seen:        make(objSet),
+		satisfied:   make(objSet),
+		dataFactory: fn,
```

Comment:
nit: informal golang conventions use the word "new" here: `newObjData` or similar:

https://golang.org/doc/effective_go.html#composite_literals